### DISCUSSION TITLE
Creating simple Blocks

### DISCUSSION THREAD

**Discussion by DerMilchmann** [BioMotionLab/TUX#27] [Q&A]
My example setup:

A single independant Variable with 5 Values which should be repeated 5 times. Easy setup, just set Trial Repetitions to 5.
Now I want to have 4 Blocks with these 25 Trials each. I could set the Experiment Repetition to 4, but itdoes not save the BlockID. The only way to I see to do this is to create another independant Variable, call it BlockID for example, and check the "Block" setting.

But now I have 2 columns called Block holding the same values, which seems kinda pointless.

Even though this works and is not a big deal to work with, I still wonder if there is a better way to do it.

**Comment by AdamBebko** [comment]
Could you be more specific about what you're using the BlockID for?  I'm not sure it's clear to me what isn't saving. Do you mean the Block Index?

**Reply by DerMilchmann** [reply]
Yes, I want to have a Block Index and the index of a Trial in a Block. Both of these exist per default and I dont know how to get them to work in a simple way

**Reply by AdamBebko** [reply]
What are you using them for? They are more for just counting which trial/block you're on in the current run, they are not really meant to be used as a data source. Maybe I can give you some ideas if I understood what the use case is

**Reply by DerMilchmann** [reply]
I want to evaluate the data per Block afterwards to see if the participant had a learning effect for example.
I can do that without the Block or Trial Index ofcourse, but I was still wondering how to get these 2 columns to work since the data looks more structured

**Reply by DerMilchmann** [reply]
Oh and Trial randomization per Block is something important I would want to use.
I am not sure if I was able to explain my problem correctly

**Comment by AdamBebko** [comment]
So are you m ore interested in comparing the repetitions then? It sounds like what you’d really like is a repetition index, which I suspect others would like as well, that’s also in the output file. Am I understanding that correctly?

**Reply by DerMilchmann** [reply]
Not exactly. I can try to show my Problem visually and record a very short video.
Words are difficult :(

**Reply by DerMilchmann** [reply]
https://www.youtube.com/watch?v=pN4LLFnd3no

**Reply by AdamBebko** [reply]
Thanks that helps a lot. So it sounds like you want the same functionality that currently exists for blocks but also for experiment repetitions, where you can chose whether trials randomize totally, within each repetition, or keep the same permutions between repetitions? If so I can look into adding this in the next version.

**Reply by DerMilchmann** [reply]
Yes I think that is it.
I want to create Blocks and access the Block features without a Block-Variable.

**Reply by AdamBebko** [reply]
Ok, I've created an issue for this feature to be included in the next version. I will be going on holidays soon though, so it might not be released until September. Thanks so much for your feedback, and helping make the project better.

**Reply by DerMilchmann** [reply]
Thank you so much for creating this framework. It's been helping me immensely :)

**Reply by AdamBebko** [reply]
sorry about the delay, still working on this!

**Reply by AdamBebko** [reply]
This is proving to be harder than expected. Not sure it will make the next version. But it's still on the list.
